tests/test-transmuxer.c

```diff
@@ -108,21 +111,69 @@ test_corrupted_empty (void)
   g_unlink (output_file);
 }
 
+static gboolean
+_generate_truncated_file (gchar * src, gchar * dst)
+{
+  gsize len, i;
+  g_autofree gchar *buffer = NULL;
+  g_autoptr (GError) error = NULL;
+
+  g_file_get_contents (src, &buffer, &len, &error);
+
+  g_assert_no_error (error);
+
+  // Use len = len / 2 to truncate file in the middle
+  len = len / 2;
+
+  // Find the last start of TS packet present in first middle part of buffer
+  for (i = len; i > 0; i--) {
+    if (buffer[i] == TS_SYNC_BYTE)
+      break;
+  }
+
+  // From the start of TS packet goes backward half TS_PACKET_SIZE bytes
+  // to have the buffer truncated in the middle of the packet
+  i -= TS_PACKET_SIZE / 2;
+
+  if (i < 0)
+    return FALSE;
+
+  g_file_set_contents (dst, buffer, i, &error);
+
+  g_assert_no_error (error);
+
+  return TRUE;
+}
+
 static void
 test_corrupted_truncated (void)
 {
   g_autoptr (HwangsaeTransmuxer) transmuxer = NULL;
   g_autoptr (GError) error = NULL;
+  g_autofree gchar *base_file = NULL;
   g_autofree gchar *output_file = NULL;
+  gchar *truncated_file = NULL;
   GSList *input_files = NULL;
   GstClockTimeDiff gap;
 
+  base_file =
+      g_test_build_filename (G_TEST_DIST, "data/test-0-5000000.ts", NULL);
+
+  truncated_file =
+      g_build_filename (g_get_tmp_dir (),
+      "transmuxer-test-XXXXXX-5000000-10000000.ts", NULL);
+
+  g_close (g_mkstemp (truncated_file), NULL);
```

Comment:
I like it. This can be an example for future contributors about how to use/handle temp files with `g_mkstemp()` simple. :)